diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/StringConcatenationRule.kt

```diff
@@ -46,19 +49,33 @@ class StringConcatenationRule : Rule("string-concatenation") {
     private fun isSingleLineStatement(node: ASTNode): Boolean =
             !node.text.contains("\n")
 
-    private fun detectStringConcatenation(node: ASTNode) {
-        node.findAllNodesWithSpecificType(BINARY_EXPRESSION).find { detectStringConcatenationInExpression(it, node) }
+    /**
+     * This method works only with top-level binary expressions. It should be checked before the call.
+     */
+    private fun detectStringConcatenation(topLevelBinaryExpr: ASTNode) {
+        val allBinaryExpressions = topLevelBinaryExpr.findAllDescendantsWithSpecificType(BINARY_EXPRESSION)
+        val nodeWithBug = allBinaryExpressions.find { detectStringConcatenationInExpression(it) }
+
+        val bugDetected = nodeWithBug != null
+
+        if (bugDetected) {
+            STRING_CONCATENATION.warnAndFix(configRules, emitWarn,
+                    this.isFixMode, topLevelBinaryExpr.text, nodeWithBug!!.startOffset) {
```

Comment:
```suggestion
                    this.isFixMode, topLevelBinaryExpr.text.lines().first(), nodeWithBug!!.startOffset) {
```
In case expression is too long and complex, I suggest to trim it for error message.